verification/spec/connection-handshake/chm.tla

```diff
@@ -0,0 +1,241 @@
+-------------------------------- MODULE CHM --------------------------------
+
+EXTENDS Naturals, FiniteSets
+
+
+CONSTANT MaxHeight     \* Maximum height of all the chains in the system.
+
+
+VARIABLES
+    chModule,   \* The state of this connection handshake module (chm).
+    turn,       \* Keep track of who takes a step: either chModule or environment.
+    inMsg,      \* A buffer holding any message incoming to the chModule.
+    outMsg      \* A buffer holding any message outbound from the chModule.
+
+
+vars == <<turn, chModule, inMsg, outMsg>>
+
+Heights == 1..MaxHeight
+
+ConnectionStates == {"UNINIT", "INIT", "TRYOPEN", "OPEN"}
+
+ConnectionIDs == {"connAtoB", "connBtoA"}
+ClientIDs == { "clientA", "clientB" }
+
+Null == "none"
+noMsg == [ type |-> "none" ]
+noErr == "errNone"
+
+
+(***************************** ConnectionEnds *****************************
+    A set of connection end records.
+    A connection end record contains the following fields:
+
+    - connectionID -- a connection identifier
+      Stores the connection identifier for this connection end.
+
+    - clientID -- a client identifier
+      Stores the client identifier associated with this connection end.
+
+    Optional, add later:
+    - prefix
+    
+ ***************************************************************************)
+ConnectionEnds ==
+    [
+        connectionID : ConnectionIDs \union {Null},
+        clientID : ClientIDs \union {Null}
+    ]
+
+
+(***************************** Connections *****************************
+    A set of connection records.
+    A connection record contains the following fields:
+
+    - localEnd -- a connection end
+      Stores the connection end for the local chain.
+
+    - remoteEnd -- a connection end
+      Stores the connection end for the remote chain.
+ ***************************************************************************)
+Connections ==
+    [
+        localEnd : ConnectionEnds,
+        remoteEnd : ConnectionEnds
+    ]
+
+
+(***************************** ConnectionHandshakeModules ******************
+    A set of records defining the CHM.
+    A CHM record contains the following fields:
+
+    - connectionState -- a string
+      Stores the current state of this connection. It has one of the
+      following values: "UNINIT", "INIT", "TRYOPEN", "OPEN".
+
+    - connection -- a connection with two ends
+      Stores the connection.
+
+    - chainHeight -- a height
+      Stores the height of the chain running alongside the CHM.
+ ***************************************************************************)
+ConnectionHandshakeModules ==
```

Comment:
Indeed, I am also confused a bit here. This is a provisional name. We should also break out the environment from the `chm.tla` file. File and module names would become clearer then.

Ideally, the CHM record should encapsulate all the information that a connection handshake module would need to carry out handshakes. We can also put all of this information as module-level fields, like the [earlier suggestion from Zarko](https://github.com/informalsystems/ibc-rs/pull/58#discussion_r412760571)?

Reply:
Not entirely sure that this would contain all the info the module needs but rather the one that it produces, i.e. the connection state. For example, the light client state is maintained by the client module and this could be part of the environment. 
Regarding the name, the spec is already called `chm`. Everything in there doesn't need again `chm`.  In the spec we know we are in the connection handshake module.
An outside module would do something like:
```
chmA == INSTANCE chm 
       WITH ConnectionIDs <- {"connAtoB"}, 
            ...
            MaxHeight <- 4
                              
queryConnection(id) == getConnection(chmA!connectionStates, id)
```